### ISSUE TITLE
Kobo Aura H20 Crash

### ISSUE THREAD

**Issue by thebombsauce** [koreader/koreader#1485]
I've had this crash twice while browsing around the file manager it freezes the kobo entirely requiring me to hold the button for 15 seconds and when I come back it has reverted all my settings to default, the crash log is empty so I don't know what to do. This is the latest developer version.

**Comment by Markismus** [comment]
Latest: koreader-nightly-20150407 aka koreader-kobo-arm-linux-gnueabihf-v2014.11-167-gded6b89.zip

I cannot reproduce this crash by browsing in the filemanager on my Kobo Aura H2O. However, I did experience some seemingly random crashes lately and the highlights of at least one document got lost  after installing this latest nightly. 

So I can confirm that the H2O is starting to become flaky. Question is whether is koreader related or KSM7update2.

Could you pinpoint the crashes? Maybe generate more information by changing [line 65 in koreader.sh](https://github.com/koreader/koreader/blob/master/platform/kobo/koreader.sh#L65):
```sh
./reader.lua $args 2> crash.log
```
to 
```sh
./reader.lua -d /mnt/onboard > koreader.log.`date +%s` 2>&1
```
?

**Comment by thebombsauce** [comment]
These two crashes in particular happened about an hour after installing KSM7update2 but I have been having crashes using koreader on this device for quite some time now (almost always when messing around in the menu).

**Comment by Markismus** [comment]
I would check your SD's: Plug into your PC>enable USB via KSM>checkdisk via Windows or Linux.

**Comment by thebombsauce** [comment]
I've had another couple of crashes, this one provided no crash log again but luckily didn't change my settings back. Seems to almost always occur when I'm playing around with the menu while reading a book.

**Comment by hwhw** [comment]
Is this with a version that has the changes introduced about 6 days ago?

**Comment by thebombsauce** [comment]
Yup this the absolute latest version (through the dev channel on ota update), I have these crashes almost every time I use it, I get kicked back to the Kobo Start Menu.

**Comment by Markismus** [comment]
I do experience more crahes to KSM now, too. `crash.log' remains empty though. It seems to happen most often when I have my reader hooked up via USB.

**Comment by chrox** [comment]
You probably would shoot on your own feet if you connect to your PC via USB while running koreader. And it's better to check the disk consistency after such crash. I also heard Kobo had more such disk problems than other devices.

**Comment by NiLuJe** [comment]
Yep, there are udev hooks that do shit in the background on USB events no matter if nickel is running or not, so it's potentially not a great idea to plug the device (even in USBNet mode).

**Comment by Markismus** [comment]
Too bad, I rather got used to hacking on koreader using WinSCP and Putty via USBNet. Would a WiFi connect give less problems?

Anyway, as the user-touch-input crashes without logs are not limited to the USB connection, I don't think it's addressed with just not using USB.

**Comment by NiLuJe** [comment]
@Markismus: That's not an issue, it's basically the only way I use KO on Kobos, you just have to be careful to not plug/unplug while something other than nickel in running.

**Comment by Markismus** [comment]
I enabled the logs, [as I suggested](https://github.com/koreader/koreader/issues/1485#issuecomment-92264805) and the logs that I generate only show turbo related crashes. (See #1498.) 
Does a log still get generated when koreader crashes to KSM? If it is always created, then these seemingly random crashes are turbo related, if not ---as in the crash.log's are empty, so the koreader.log's, too--- we have two problems.
